### PR TITLE
FLEXIN-329: Added changes to run e2e in other regions

### PR DESCRIPTION
**Pull request by ashishkumarTWLO**
Added the required changes to run e2e in different region/env

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: cypress/plugins/helpers/interactionHandler.ts [hunk 1]
```diff
@@ -161,7 +165,7 @@ export const completeReservation = async ({ conversationSid }: { conversationSid
 };
 
 export const getCustomerName = async ({ conversationSid }: { conversationSid: string }) => {
-    await new Promise((res) => setTimeout(res, 2000)); // Add buffer to avoid api calls being made too close together
+    await new Promise((res) => setTimeout(res, 5000)); // Add buffer to avoid api calls being made too close together
```
Comment (AnudeepChPaul) [inline cypress/plugins/helpers/interactionHandler.ts:168]: @ashishkumarTWLO we did this for testing. Let's revert back to 2000.

File: cypress/plugins/helpers/interactionHandler.ts [hunk 2]
```diff
@@ -179,38 +183,40 @@ export const getLastMessageMediaData = async ({ conversationSid }: { conversatio
...
     const messageIM = await client.conversations
         .conversations(conversationSid)
         .messages.list({ order: "desc", limit: 1 });
     const mediaSid = messageIM[0].media[0].sid;
+
+    const conversationInstance = await client.conversations.conversations(conversationSid).fetch();
     const options = {
         method: "GET",
-        url: `https://mcs.us1.twilio.com/v1/Services/ISefe4b42a882440c0adcfd843a197151a/Media/${mediaSid}`,
+        url: `https://mcs.${parseRegionForConversations}twilio.com/v1/Services/${conversationInstance.chatServiceSid}/Media/${mediaSid}`,
```
Comment (AnudeepChPaul) [inline cypress/plugins/helpers/interactionHandler.ts:211]:
```suggestion
        url: `https://mcs.${parseRegionForConversations(process.env.REACT_APP_REGION)}twilio.com/v1/Services/${conversationInstance.chatServiceSid}/Media/${mediaSid}`,
```

**Review by AnudeepChPaul — CHANGES_REQUESTED** [review]

File: cypress/plugins/helpers/interactionHandler.ts (hunk 1 above)
Comment (ashishkumarTWLO) [reply cypress/plugins/helpers/interactionHandler.ts]: done

File: cypress/plugins/helpers/interactionHandler.ts (hunk 2 above)
Comment (ashishkumarTWLO) [reply cypress/plugins/helpers/interactionHandler.ts]: done

**Review by AnudeepChPaul — APPROVED** [review]
Approved!

File: cypress.config.ts [hunk 3]
```diff
@@ -24,6 +26,18 @@ export default defineConfig({
         trashAssetsBeforeRuns: true,
         responseTimeout: 100000,
         setupNodeEvents(on, config) {
+            on("before:browser:launch", async () => {
+                const client = getTwilioClient();
```
Comment (vinesh-kumar) [inline cypress.config.ts:30]: can we add some comments here explaining what we do here and why, if possible add ticket number, slack message links etc

File: cypress/plugins/helpers/interactionHandler.ts [hunk 4]
```diff
@@ -103,7 +107,7 @@ export const acceptReservation = async ({ conversationSid }: { conversationSid:
 };
 
 export const sendMessage = async ({ conversationSid, messageText }) => {
-    const client = await getTwilioClient();
+    const client = getTwilioClient();
```
Comment (vinesh-kumar) [inline cypress/plugins/helpers/interactionHandler.ts:110]: was this incorrect usage before?

File: cypress.config.ts (hunk 3 above)
Comment (ashishkumarTWLO) [reply cypress.config.ts:31]: while testing we noticed sometimes worker is occupied with task from previous run and was not able to accept the new ones, so to fix this, we are removing old tasks every new run. Added the comment as well

File: cypress/plugins/helpers/interactionHandler.ts (hunk 4 above)
Comment (ashishkumarTWLO) [reply cypress/plugins/helpers/interactionHandler.ts:110]: yes, this function is not `async`, so there was not need for the keyword `await`

**Review by vinesh-kumar — APPROVED** [review]

**Pull request merged**